### PR TITLE
support to upload file

### PR DESCRIPTION
**Pull request by takaishi**
Upload file to slack with Faraday::UploadIO.

``` ruby
arg = {
    channels: '#foo',
    file: Faraday::UploadIO.new('bar.txt', 'text/plain')
}
Breacan.files_upload(arg)
```

### CODE REVIEW

**Comment by takaishi** [comment]
I forgot to fix sawyer...

File: lib/breacan/client/files.rb [hunk 1]
```diff
@@ -14,8 +15,35 @@ def files_list(args)
...
+        reset_files_upload_builder
+        res
+      end
+
+      def set_files_upload_builder
+        @serializer = Breacan::Client::Files::FileSerializer.any_json
+        @middleware = Faraday::RackBuilder.new do |c|
+          c.request :multipart
+          c.request :url_encoded
+          c.adapter :net_http
```
Comment (linyows) [inline lib/breacan/client/files.rb:35]: I think I had better in the same format as the other response.
https://github.com/linyows/breacan/pull/10/files#diff-938daba010f4fc9bda8d4e08d8ef6758R68

File: lib/breacan/client/files.rb [hunk 2]
```diff
@@ -14,8 +15,35 @@ def files_list(args)
...
+        end
+      end
+
+      def reset_files_upload_builder
+        @middleware = default_builder
+      end
+
+      class FileSerializer < Sawyer::Serializer
+        def encode(data)
+          data
```
Comment (linyows) [inline lib/breacan/client/files.rb:46]: hmmm

Comment (takaishi) [inline lib/breacan/client/files.rb:46]: Sawer's default serializer encode UploadIO object to String...

https://github.com/lostisland/sawyer/blob/master/lib/sawyer/serializer.rb#L49

**Comment by linyows** [comment]
Looks good, Thank you!

**Comment by linyows** [comment]
This was released as v0.7.0.

**Pull request merged**